src/device/mod.rs

```diff
@@ -689,6 +689,84 @@ impl RequestedState {
             })
             .collect()
     }
+
+    // Same as collect_dns_records() fn but for peers with defined nicknames.
+    pub fn collect_dns_nickname_records(&self) -> Records {
+        let mut seen_nicknames = HashSet::new();
+
+        self.meshnet_config
+            .clone()
+            .map_or(Vec::new(), |cfg| {
+                [
+                    cfg.peers.map_or(Vec::new(), |peers| peers),
+                    vec![Peer {
+                        base: cfg.this,
+                        ..Default::default()
+                    }],
+                ]
+                .concat()
+            })
+            .iter()
+            .filter_map(|v| {
+                v.nickname.as_ref().and_then(|nick| {
+                    if seen_nicknames.insert(nick) {
+                        if self.validate_record(nick) {
+                            v.ip_addresses
+                                .as_ref()
+                                .map(|ips| (nick.to_owned(), ips.clone()))
+                        } else {
+                            None
+                        }
+                    } else {
+                        telio_log_warn!("Nickname is already assigned: {:?}. Ignore", nick);
+                        None
+                    }
+                })
+            })
+            .map(|(nickname, ip)| (format!("{}.nord", nickname).to_owned(), ip.clone()))
+            .collect()
+    }
+
+    /// Validate records (RFC #009)
+    fn validate_record(&self, name: &String) -> bool {
```

Comment:
API should be doing nickname validation, and probably also by the client for better UX. Thus, this checks might be irrelevant/not needed.